packages/core-js-builder/index.d.ts

```diff
@@ -0,0 +1,33 @@
+type Summary = {
+    size: boolean,
+    modules: boolean
+};
+
+type StringOrRegExp = string | RegExp;
+
+type MinEnvTarget = Record<string, string> & { esmodules?: boolean };
```

Comment:
It's a little more complex. A record can contain string, boolean, or array values. See, for example, Babel typing in a similar case https://github.com/babel/babel/blob/d3bce24473c5efef1a80746bdb7ee615bb29f778/packages/babel-helper-compilation-targets/src/types.d.ts

Reply:
For array I could find a reference within the source code (fixed this!), but I only see one boolean, the `esmodules` option.

Reply:
Yes, and `esmodules` can be mixed with other targets, and that's not allowed with the current typing.

Reply:
Should word now with `Record<string, string | readonly string[]> | { esmodules?: boolean };`